pandas/io/pytables.py

```diff
@@ -2453,19 +2456,22 @@ def set_atom_categorical(self, block):
 
         values = block.values
         codes = values.codes
-        self.kind = "integer"
+
         self.dtype = codes.dtype.name
         if values.ndim > 1:
             raise NotImplementedError("only support 1-d categoricals")
 
+        assert self.dtype.startswith("int"), self.dtype
+
```

Comment:
Not sure I follow the motivation for this.  Seems out of scope.

Reply:
making sure that i understand correctly that the codes is always signed-integer